hoomd/HOOMDMath.cc

```diff
@@ -61,7 +61,7 @@ void export_hoomd_math_functions(pybind11::module& m)
             "w",
             [](const double4& s) { return (double)s.w; },
             [](double4& s, double v) { s.w = v; });
-#else
+
     pybind11::class_<float2, std::shared_ptr<float2>>(m, "float2")
         .def(pybind11::init<>())
         .def_property(
```

Comment:
Or any of the `char` or `uint`. We use `int3` in `box.py` but given you deleted `Scalar3` this should be fine to remove as well.

Reply:
Python and pybind11 do not understand typedefs - they see only the base type. Previously, the code exported only `double[N]` or `float[N]` depending on the preprocessor definition `SINGLE_PRECISON`. Now the code always exports both. I made this change so that any future code that interoperates with Python using `ShortReal[N]` would function when `HOOMD_SHORTREAL_SIZE=32`.

You are correct that many of these are unused. They are leftover from HOOMD v1 and v2 where the Python interface layer worked more closely with the native C++ types. Now (in many cases) we provide specialized methods for Python that work with `pybind::list` and `pybind11::tuple` in place of these.

I removed all the exports I could without failing tests. It appears that `Box` may be the last location in HOOMD where Python directly works with these C++ vector types. In a future PR we could refactor the Box exports so that uses tuples instead.

Reply:
[`uint3` is required as well](https://github.com/glotzerlab/hoomd-blue/actions/runs/3621200364/jobs/6104594641#step:6:2807).